app/screens/post_options/post_options.js

```diff
@@ -110,19 +111,27 @@ export default class PostOptions extends PureComponent {
     }
 
     getFollowThreadOption = () => {
-        const {thread} = this.props;
-        if (!thread) {
+        const {location, thread} = this.props;
+        if (location !== CHANNEL) {
```

Comment:
So cannot follow a thread from the channel screen?

Reply:
You can follow/unfollow a thread/message by long pressing the posts on the channel screen.